src/domain/index.ts

```diff
@@ -9,6 +9,31 @@ export enum EthereumEvent {
   DISCONNECT = "disconnect",
 }
 
+export type TransactionStatus = "processing" | "initiated" | "on-hold" | "completed" | "failed";
+
+export enum TransactionStatusText {
+  PROCESSING = "Processing",
+  INITIATED = "Initiated",
+  ON_HOLD = "On Hold",
+  COMPLETED = "Completed",
+  FAILED = "Error",
+}
+
+export function transactionStatusText(status: TransactionStatus): TransactionStatusText {
```

Comment:
Now that I see this again, I think that it would make more sense to make it a React component. We can delete the enum as well if we move it to a component.

Reply:
Enum removed. Only function left.